agent/Retriever.py

```diff
@@ -0,0 +1,91 @@
+from llama_index import (
+    VectorStoreIndex,
+    SummaryIndex,
+    SimpleDirectoryReader,
+    ServiceContext
+)
+from llama_index.agent import OpenAIAgent
+from llama_index.schema import IndexNode
+from llama_index.tools import QueryEngineTool, ToolMetadata
+from llama_index.llms import OpenAI
+from llama_index.retrievers import RecursiveRetriever
+from llama_index.query_engine import RetrieverQueryEngine
+from llama_index.response_synthesizers import get_response_synthesizer
+
+
+class Retriever:
+    """
+    A class to handle the retrieval and querying of documents using Llama Index services.
+
+    Attributes:
+        llm (OpenAI): An instance of the OpenAI model.
+        service_context (ServiceContext): The context for the service.
+        docs (dict): A dictionary holding the loaded documents.
+        query_engine (RetrieverQueryEngine): The query engine for retrieval tasks.
+    """
+
+    def __init__(self, index_path):
+        """
+        Initialize the Retriever with a specified index path.
+
+        Args:
+            index_path (str): The path to the index directory.
+        """
+        self.llm = OpenAI(temperature=0, model="gpt-4-1106-preview")
+        self.service_context = ServiceContext.from_defaults(llm=self.llm)
+        self.docs = {}
+        self.query_engine = None
+
+    def load_documents(self, files):
+        """
+        Load documents from the given file paths.
+
+        Args:
+            files (list): A list of file paths to load.
+        """
+        for file in files:
+            self.docs[file] = SimpleDirectoryReader(
+                input_files=[f"./{file}"]
```

Comment:
Using String formatting for file paths can be dangerous if not validated correctly because of potential 
    security issues. Make sure to validate or sanitize the input paths or, if the files are always in the current directory, 
    you can omit the string formatting.
```suggestion
    self.docs[file] = SimpleDirectoryReader(input_files=[file]).load_data()
```